includes/object-cache.php

```diff
@@ -702,23 +702,26 @@ protected function connect_using_phpredis( $parameters ) {
                 'retry_interval' => (int) $parameters['retry_interval'],
             ];
 
+            if ( version_compare( $version, '3.1.3', '>=' ) ) {
+                $args['read_timeout'] = $parameters['read_timeout'];
+            }
+
             if ( strcasecmp( 'tls', $parameters['scheme'] ) === 0 ) {
                 $args['host'] = sprintf(
                     '%s://%s',
                     $parameters['scheme'],
                     str_replace( 'tls://', '', $parameters['host'] )
                 );
+                if ( version_compare( $version, '5.3.0', '>=' ) && defined( 'WP_REDIS_TLS' ) && !empty(WP_REDIS_TLS) ) {
+                    $args['others']['stream'] = WP_REDIS_TLS;
+                }
```

Comment:
This looks good. You can do the same code for `connect_using_relay()`, the API is identical.

Reply:
At what version of Relay does the stream option become supported? I can see that you maintain the package. Is ir [0.6.2](https://github.com/cachewerk/relay/releases/tag/v0.6.2) where you mention changing the arity of connect()?

Reply:
You can skip the version check for Relay. Most of them support it.